task.sql

```diff
@@ -1,17 +1,40 @@
 CREATE DATABASE ShopDB; 
 USE ShopDB; 
 
--- Create a table to store countries 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 ) ENGINE=InnoDB;
 
--- Create a table for caching GeoIP data (Columns: ID, IP Range, CountryID)
+CREATE TABLE GeoIPCache (
+    ID INT AUTO_INCREMENT,
+    IPRange VARCHAR(50),
+    CountryID INT,
+    
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries (ID) ON DELETE CASCADE
+) ENGINE=Memory;
 
--- Create a table for storing product descriptions for different countries (Columns: ID, CountryID, ProductID, Description )
+CREATE TABLE ProductDescription (
+    ID INT AUTO_INCREMENT,
+    CountryID INT,
+    ProductID INT,
+    Description TEXT,
+    
+    PRIMARY KEY (ID)
+) ENGINE=MyISAM;
+
+CREATE TABLE Logs (
+    ID INT AUTO_INCREMENT,
+    Timestamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    Message TEXT,
 
--- Create a table for storing logs. For now we don't need to save them, but we need to implement functionality (Columns: ID, Time, LogRecord)
+    PRIMARY KEY (ID)
+) ENGINE=Blackhole;
```

Comment:
The `Logs` table uses the `Blackhole` storage engine, which does not store data. Ensure this is intentional, as it will not retain any log entries.